src/context/auth.context.tsx

```diff
@@ -0,0 +1,83 @@
+import { createContext, ReactNode, useEffect, useState } from "react";
+import { Customer } from "../interfaces/customers/customer.interface";
+import { useFetchCustomer } from "../hooks/customers/useFetchCustomer";
+
+export interface IAuthContext {
+  customer: Customer | null;
```

Comment:
Let's name it user even if in our design is a Customer type. This in general is different but in our app the Customer also had details about role and so on.